src/plugins/visualize/public/application/utils/use/use_editor_updates.ts

```diff
@@ -84,15 +85,17 @@ export const useEditorUpdates = (
       });
 
       const handleLinkedSearch = (linked: boolean) => {
-        if (linked && !savedVis.savedSearchId && savedSearch) {
+        if (linked && savedVis && !savedVis.savedSearchId && savedSearch) {
           savedVis.savedSearchId = savedSearch.id;
           vis.data.savedSearchId = savedSearch.id;
           if (vis.data.searchSource) {
             vis.data.searchSource.setParent(savedSearch.searchSource);
           }
-        } else if (!linked && savedVis.savedSearchId) {
+        } else if (!linked && savedVis && savedVis.savedSearchId) {
           delete savedVis.savedSearchId;
           delete vis.data.savedSearchId;
+        } else {
+          // TODO: something to do for when it's not a saved vis?
```

Comment:
Should it be resolved before merging?
Unlinking isn't working right now in `edit_by_value` mode.

![unlink](https://user-images.githubusercontent.com/31325372/88195525-d024b680-cc48-11ea-884e-b5300e72035e.gif)

Reply:
I'd really like to keep this PR as small as possible and build on it incrementally. I believe for the first pass it can work without saved search linking/unlinking.